SM-60 - Create endpoint to to retrieve a material for reparation request

JSTLWK opened the pull request:
# SM-60 - Create endpoint to to retrieve a material for reparation request

**Changelogs**
- Added gitignore files
- Added packages for filters and sorting and json pagination.
- Created Index for all material reparations
- Edited env for testing; we should pick this up.


**Notes**
Regarding the .env.testing; we should find a way to use both methods to test.  We should either gitignore phpunit.xml  or .env.testing and write some documentation on the information in the file.

We will decide tomorrow if we can. For now, I will keep this way of working.

JSTLWK committed:
SM-60 - Create endpoint to to retrieve a material for reparation request

- Added gitignore files
- Added packages for filters and sorting and json pagination.
- Created Index for all material reparations
- Edited env for testing; we should pick this up.

Commit:
style: fix code style

cyrildewit commented on .env.testing:
```diff
@@ -2,24 +2,24 @@ APP_NAME="KA-Huis Testing"
 APP_KEY=base64:p8lPFTEgvQ+FcmW0jetdW/9MIwnigmiJUjxJe1eiebs=
 APP_DEBUG=true
-APP_URL=https://space-management.localhost
+APP_URL=http://space-management.test/
```
These changes shouldn't be commited, but I saw that you already addressed an issue with this file in your description, so we will talk about it in our meeting todayl

cyrildewit commented on config/app.php:
```diff
@@ -167,6 +167,8 @@
         /*
          * Package Service Providers...
          */
+        Spatie\JsonApiPaginate\JsonApiPaginateServiceProvider::class,
```
These aren't necessary, because they're auto discovered by the framework.

cyrildewit commented on database/migrations/2022_04_14_101120_create_sessions_table.php:
```diff
@@ -42,6 +42,6 @@ public function down(): void
     private function shouldRun(): bool
     {
-        return $this->application->environment('acceptance');
+        return $this->application->environment(['acceptance', 'local', 'testing']);
```
Why don't we use redis? There is already a container available.

cyrildewit commented on phpunit.xml:
```diff
@@ -12,23 +12,20 @@
         <include>
             <directory suffix=".php">./app</directory>
         </include>
-        <report>
```
Why did you remove this? I think it's necessary for CodeCov.io.

cyrildewit commented on phpunit.xml:
```diff
@@ -12,23 +12,20 @@
+        <server name="MAIL_MAILER" value="array"/>
+        <server name="QUEUE_CONNECTION" value="sync"/>
+        <server name="SESSION_DRIVER" value="array"/>
+        <server name="TELESCOPE_ENABLED" value="false"/>
```
Telescope is not installed, so this can be removed.

cyrildewit commented on phpunit.xml:
```diff
@@ -12,23 +12,20 @@
+        <server name="BCRYPT_ROUNDS" value="4"/>
+        <server name="CACHE_DRIVER" value="array"/>
+        <server name="DB_CONNECTION" value="mysql"/>
+        <server name="DB_DATABASE" value="spacemangement_dev"/>
```
I thin this will break the docker setup.

cyrildewit reviewed (changes requested):
See comments

JSTLWK committed:
SM-60 - Pull request changes

- reverted some committed changes.

Commit:
style: fix code style

cyrildewit commented on app/Http/Controllers/API/V1/ReparationRequestMaterialController.php:
```diff
@@ -0,0 +1,37 @@
+{
+    public function index(Request $request): LengthAwarePaginator
+    {
+        $materialRequest = QueryBuilder::for(ReparationRequestMaterial::class)
```
`$materialRequest` is a little bit confusing. I thinks `$material` or the full out version `$reparationRequestMaterial` is more specific.

cyrildewit commented on database/migrations/2020_04_14_101120_create_cache_table.php:
```diff
@@ -0,0 +1,43 @@
+
+    private function shouldRun(): bool
+    {
+        return $this->application->environment(['acceptance', 'local', 'testing']);
```
We could eliminate the need for this migration on the local and testing environment if everyone uses the docker setup locally.

JSTLWK commented:
Add changes regarding Ticket #55

cyrildewit commented on app/Http/Controllers/API/V1/ReparationRequestMaterialController.php:
```diff
@@ -0,0 +1,37 @@
+            ])
+            ->jsonPaginate();
+
+        return $materialRequest;
```
We agreed on using specific resource classed right? Then this should be picked up.

cyrildewit replied:
We agreed on using specific resource classed right? Then this should be picked up.

cyrildewit reviewed (changes requested):
See comments

JSTLWK committed:
Merge branch 'acceptance' into feature/SM-60

# Conflicts:
#	.php-cs-fixer.cache
#	composer.json
#	composer.lock
#	routes/api.php

Commit:
style: fix code style

JSTLWK committed:
SM-60: Create endpoint to retrieve materials - Pull request feedback

This is a combination of both commented feedback and feedback we have discussed from the other pull request.

Commit:
style: fix code style

JSTLWK committed:
SM-60: Create endpoint to retrieve materials - Pull request feedback

This is a combination of both commented feedback and feedback we have discussed from the other pull request.

JSTLWK commented:
The following commit has the changes:

# SM-60: Create endpoint to retrieve materials - Pull request feedback
This is a combination of both commented feedback and feedback we have discussed from the other pull request.

Could you please review the pull request @cyrildewit?

JSTLWK committed:
SM-60: Create endpoint to retrieve materials - Pull request feedback

This is a combination of both commented feedback and feedback we have discussed from the other pull request.

Commit:
style: fix code style

cyrildewit commented on routes/api/v1.php:
```diff
@@ -17,4 +17,11 @@
             });
+
+        $router
+            ->prefix('reparation_requests_material')
```
I think the convention is usually to use the plural form. I've used it too. For example: https://stripe.com/docs/api/terminal/locations/update

cyrildewit replied:
```suggestion
            ->prefix('reparation_request_materials')
```

cyrildewit committed:
SM-60 - Change route path

cyrildewit committed:
style: fix code style

cyrildewit reviewed (approved):
LGTM
